File: BaseBackend/src/main/java/com/common/model/sql/Tenant.java
package com.common.model.sql;

import com.common.model.enumerate.EntityStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import lombok.experimental.FieldDefaults;
import lombok.experimental.SuperBuilder;

/**
 * Entity đại diện cho Công ty / Tổ chức (Tenant Multitenancy).
 *
 * @created_at 2026-07-31
 * @author txhoan
 */
@Entity
@Table(name = "sys_tenants")
@Getter
@Setter
@ToString(callSuper = true)
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Tenant extends BaseSoftDeleteModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column(nullable = false, length = 100)
    String name;

    @Column(nullable = false, unique = true, length = 50)
    String code;

    @Column(length = 255)
    String domain;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @lombok.Builder.Default
    EntityStatus status = EntityStatus.ACTIVE;
}

File: be/.Net/Swe.Common/Model/Tenant.cs
using System.ComponentModel.DataAnnotations;
using Swe.Common.Attributes;
using Swe.Common.Base;
using Swe.Common.Enum;

namespace Swe.Common.Model;

/// <summary>
/// Entity đại diện cho Công ty / Tổ chức (Tenant Multitenancy).
/// </summary>
/// <created_at>2026-08-03</created_at>
/// <author>txhoan</author>
[ConfigTable("sys_tenants")]
public class Tenant : BaseSoftDeleteModel
{
    [Key]
    [ConfigColumn("id")]
    public long Id { get; set; }

    [Required]
    [MaxLength(100)]
    [ConfigColumn("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(50)]
    [ConfigColumn("code")]
    [CheckDuplicate("DuplicatedTenantCode")]
    public string Code { get; set; } = string.Empty;

    [MaxLength(255)]
    [ConfigColumn("domain")]
    public string? Domain { get; set; }

    [Required]
    [ConfigColumn("status")]
    public string Status { get; set; } = AppEnum.EntityStatus.ACTIVE.ToString();
}
